sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -1350,45 +1336,23 @@ class CudfRegexTranspiler(mode: RegexMode) {
               // when the previous character is a line anchor ($), the JVM has special handling
               // when matching against line terminator characters
               case Some(RegexChar('$')) | Some(RegexEscaped('Z')) =>
-                val j = r.lastIndexWhere {
-                  case RegexEmpty() => false
-                  case _ => true
-                }
                 part match {
                   case RegexGroup(capture, RegexSequence(
                       ListBuffer(RegexCharacterClass(true, parts))), _)
                       if parts.forall(!isBeginOrEndLineAnchor(_)) =>
-                    r(j) = RegexSequence(
-                      ListBuffer(lineTerminatorMatcher(excludeCRLF = true, capture = capture),
-                        RegexChar('$')))
                     popBackrefIfNecessary(capture)
                   case RegexGroup(capture, RegexCharacterClass(true, parts), _)
                       if parts.forall(!isBeginOrEndLineAnchor(_)) =>
-                    r(j) = RegexSequence(ListBuffer(
-                      lineTerminatorMatcher(excludeCRLF = true, capture = capture),
-                        RegexChar('$')))
                     popBackrefIfNecessary(capture)
                   case RegexCharacterClass(true, parts)
                       if parts.forall(!isBeginOrEndLineAnchor(_)) =>
-                    r(j) = RegexSequence(ListBuffer(
-                      lineTerminatorMatcher(excludeCRLF = true, capture = false),
-                        RegexChar('$')))
                     popBackrefIfNecessary(false)
-                  case RegexChar(ch) if ch == '\n' =>
+                  case RegexChar(ch) if "\n\r\u0085\u2028\u2029".contains(ch) =>
```

Comment:
is this lineTerminatorChars, but concated together? Could we make this unified? For both maintainability but also to make it more readable?

Reply:
Done, thanks for pointing it out!